components/core/dock.tsx

```diff
@@ -0,0 +1,220 @@
+};
+
+type DocContextType = {
+  mouseX: MotionValue;
+  spring: SpringOptions;
+  magnification: number;
+  distance: number;
+};
+type DockProviderProps = {
+  children: React.ReactNode;
+  value: DocContextType;
+};
+/** End of types */
+
+/** Context */
+const DockContext = createContext<DocContextType | undefined>(undefined);
+
+function DockProvider({ children, value }: DockProviderProps) {
+  return <DockContext.Provider value={value}>{children}</DockContext.Provider>;
+}
+/** End of context */
+
+/** Hooks */
+function useDock() {
+  const context = useContext(DockContext);
+  if (!context) {
+    throw new Error('useDock must be used within an DockProvider');
+  }
+  return context;
+}
+/** End of hooks */
+
+function Dock({
+  children,
+  className,
+  spring = { mass: 0.1, stiffness: 150, damping: 12 },
+  magnification = DEFAULT_MAGNIFICATION,
+  distance = DEFAULT_DISTANCE,
+}: TDock) {
+  const mouseX = useMotionValue(Infinity);
+  const isHovered = useMotionValue(0);
+
+  const height = useTransform(
+    isHovered,
+    [0, 1],
+    ['auto', Math.max(DOCK_HEIGHT, magnification + magnification / 2 + 4)]
+  );
+
+  return (
+    <motion.div
+      style={{
+        height: height,
+        scrollbarWidth: 'none',
+      }}
+      className='mx-2 flex max-w-full items-end overflow-x-auto'
+    >
+      <motion.div
+        onMouseMove={({ pageX }) => {
+          isHovered.set(1);
+          mouseX.set(pageX);
+        }}
+        onMouseLeave={() => {
+          isHovered.set(0);
+          mouseX.set(Infinity);
+        }}
+        className={cn(
+          'mx-auto flex h-16 w-fit items-end gap-4 rounded-2xl bg-gray-50 px-4 pb-3 dark:bg-neutral-900',
+          className
+        )}
+        role='toolbar'
+        aria-label='Application dock'
+      >
+        <DockProvider value={{ mouseX, spring, distance, magnification }}>
+          {children}
+        </DockProvider>
+      </motion.div>
+    </motion.div>
+  );
+}
+
+function DockItem({ children, className }: TDockItem) {
+  const ref = useRef<HTMLDivElement>(null);
+
+  const { distance, magnification, mouseX, spring } = useDock();
+
+  // To control hovered animation for label
+  const isHovered = useMotionValue(0);
+
+  const mouseDistance = useTransform(mouseX, (val) => {
+    const domRect = ref.current?.getBoundingClientRect() ?? { x: 0, width: 0 };
+    return val - domRect.x - domRect.width / 2;
+  });
+
+  const widthTransform = useTransform(
+    mouseDistance,
+    [-distance, 0, distance],
+    [40, magnification, 40]
+  );
+
+  const width = useSpring(widthTransform, spring);
+
+  return (
+    <motion.div
+      ref={ref}
+      style={{ width }}
+      onHoverStart={() => isHovered.set(1)}
+      onHoverEnd={() => isHovered.set(0)}
+      onFocus={() => isHovered.set(1)}
+      onBlur={() => isHovered.set(0)}
+      className={cn(
+        'relative inline-flex aspect-square items-center justify-center rounded-full bg-gray-200 dark:bg-neutral-800',
```

Comment:
same, move `inline-flex aspect-square items-center justify-center rounded-full bg-gray-200 dark:bg-neutral-800` to the `AppleStyleDock` example